store/store.go

```diff
@@ -467,6 +462,17 @@ func (s *Store[H]) flush(ctx context.Context, headers ...H) error {
 	return batch.Commit(ctx)
 }
 
+func (s *Store[H]) loadHeadKey(ctx context.Context) error {
```

Comment:
lets godoc it as well

Reply:
Confused about the name. It doesn't load the `headKey` but the head header.

Reply:
we also have one type of head now, so we should ideally be consistent with naming, which means it should be `contiguousHead` or just `head` on all the methods. As the head is now contiguous by default(actually previously it was contiguous as well), I would do the latter.

Reply:
Changed to `loadContiguousHead`. Done.